scripts/native_simulator/Makefile

```diff
@@ -60,7 +60,7 @@ NSI_OPT?=-O0
 #  Warnings switches (for the runner itself)
 NSI_WARNINGS?=-Wall -Wpedantic
 #  Preprocessor flags
-NSI_CPPFLAGS?=-D_POSIX_C_SOURCE=200809 -D_XOPEN_SOURCE=600 -D_XOPEN_SOURCE_EXTENDED
+NSI_CPPFLAGS?=-D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=600 -D_XOPEN_SOURCE_EXTENDED
```

Comment:
Just a sidenote: This one is not mingled with Zephyr's embedded definitions, so you should not need to change it. (it does not matter much if you do though)

Reply:
This fix aligns the code with the POSIX standard where the constants are expected to be of type long, not int. And this allows us to avoid `#undef` for code which explicitly needs POSIX functionality as the values will match.